testkit/scaladsl/src/main/scala/com/lightbend/lagom/scaladsl/testkit/ServiceTest.scala

```diff
@@ -242,43 +245,15 @@ object ServiceTest {
     val now = DateTimeFormatter.ofPattern("yyMMddHHmmssSSS").format(LocalDateTime.now())
     val testName = s"ServiceTest_$now"
 
-    val log = LoggerFactory.getLogger(getClass)
-
     val lifecycle = new DefaultApplicationLifecycle
 
     val config =
       if (setup.cassandra) {
-        val cassandraPort = CassandraLauncher.randomPort
-        val cassandraDirectory = Files.createTempDirectory(testName)
-
-        // Shut down Cassandra and delete its temporary directory when the application shuts down
-        lifecycle.addStopHook { () =>
-          import scala.concurrent.ExecutionContext.Implicits.global
-          Try(CassandraLauncher.stop())
-          // The ALLOW_INSECURE option is required to remove the files on OSes that don't support SecureDirectoryStream
-          // See http://google.github.io/guava/releases/snapshot-jre/api/docs/com/google/common/io/MoreFiles.html#deleteRecursively-java.nio.file.Path-com.google.common.io.RecursiveDeleteOption...-
-          Future(MoreFiles.deleteRecursively(cassandraDirectory, RecursiveDeleteOption.ALLOW_INSECURE))
-        }
-
-        val t0 = System.nanoTime()
-
-        CassandraLauncher.start(
-          cassandraDirectory.toFile,
-          LagomTestConfigResource,
-          clean = false,
-          port = cassandraPort,
-          CassandraLauncher.classpathForResources(LagomTestConfigResource)
-        )
-
-        log.debug(s"Cassandra started in ${
-          TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - t0)
-        } ms")
-
-        Configuration(TestUtil.persistenceConfig(testName, cassandraPort, useServiceLocator = false)) ++
-          Configuration("lagom.cluster.join-self" -> "on")
-
-      } else if (setup.cluster || setup.jdbc) {
-        Configuration("lagom.cluster.join-self" -> "on")
+        configureCassandra(testName, lifecycle)
```

Comment:
Great! Much cleaner like that.

Reply:
This is a lot cleaner.

Just a small nitpick, I'd move the `++ ClusterConfiguration` from `configureCassandra(...)`'s last line to here so it looks like: (including a reordering)

```scala
 val config =
      if (setup.cassandra) {
        ClusterConfiguration ++ configureCassandra(testName, lifecycle)
      } else if (setup.jdbc) {
        ClusterConfiguration ++ JdbcConfiguration
      } else if (setup.cluster) {
        ClusterConfiguration
      } else {
        Configuration.empty
      }
```

Reply:
Actually, from my comment in the equivalent code for `javadsl`, I think this could even be:

```scala
 val config =
      if (setup.cassandra) {
        val cassandraPort = runCassandraServer(...)  // not sure what arguments
        val cassandraConfig = Configuration(TestUtil.persistenceConfig(testName, cassandraPort, useServiceLocator = false))
        ClusterConfiguration ++ cassandraConfig
      } else if (setup.jdbc) {
        ClusterConfiguration ++ JdbcConfiguration
      } else if (setup.cluster) {
        ClusterConfiguration
      } else {
        Configuration.empty
      }
```

Reply:
Yes, I agree, @ignasi35. I'll line up the two implementations.